### PR TITLE
[3.5] Backport livez/readyz

### PR DESCRIPTION
**Pull request by siyuanfoundation**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Part of https://github.com/etcd-io/etcd/issues/16007

Follow up of https://github.com/etcd-io/etcd/pull/17000

Original PRs:
https://github.com/etcd-io/etcd/pull/16697
https://github.com/etcd-io/etcd/pull/16651
https://github.com/etcd-io/etcd/pull/16797
https://github.com/etcd-io/etcd/pull/16984

### CODE REVIEW

**Comment by serathius** [comment]
Should we backport e2e tests too?

**Comment by siyuanfoundation** [comment]
In order to backport the e2e test, it would require backporting a lot of the e2e test framework, because existing Failpoint is in a this agent handler  which is very hard to reuse. Is that worth the effort?

**Comment by serathius** [comment]
Yes, it requires some effort, but we should not add features without proper e2e test coverage. Please let me know if you need help with backport.

**Comment by serathius** [comment]
Note, I didn't mean that we should backport the e2e framework changes, just backport the e2e test.

**Comment by serathius** [comment]
Looking at the e2e framework backports I think it would be easier to move it to separate PR so we ensure it's properly reviewed.

**Review by serathius — APPROVED** [review]

**Comment by ahrtr** [comment]
Also which PR are you backporting? Please provide a link to the original PR.

File: server/etcdserver/api/etcdhttp/health.go [hunk 1]
```diff
@@ -12,15 +12,22 @@
...
 
 import (
+	"bytes"
 	"context"
 	"encoding/json"
 	"fmt"
 	"net/http"
 	"time"
 
+	"path"
```
[the remark this answers is not in the document]
Comment (siyuanfoundation) [reply server/etcdserver/api/etcdhttp/health.go:26]: Removed

**Pull request merged**